api/v1/agentconfig_types.go

```diff
@@ -311,6 +315,14 @@ func (c AgentConfigSpecAdapter) GetInstallationServiceAccount() string {
 	return c.original.InstallationServiceAccount
 }
 
+// SetRetryAnnotation flags the resource to retry its last operation.
+func (c *AgentConfigSpecAdapter) GetRetryLimit() (int, bool) {
```

Comment:
If this function returned a *Int32, we could assign it directly to the JobSpec without as many checks and conditions in the controller. Is using a bool here instead of nil is providing value over returning a single *Int32?